packages/flutter_tools/lib/src/device.dart

```diff
@@ -135,12 +132,13 @@ class DeviceManager {
   }
 
   /// Return the list of all connected devices.
-  Stream<Device> getAllConnectedDevices() async* {
-    for (final DeviceDiscovery discoverer in _platformDiscoverers) {
-      for (final Device device in await discoverer.devices) {
-        yield device;
-      }
-    }
+  Future<List<Device>> getAllConnectedDevices() async {
```

Comment:
@zanderso in https://github.com/flutter/flutter/pull/49854#discussion_r373673837:
> This could return a `Stream` if we're actually getting results bit-by-bit [...] or if this is implementing an interface that requires it to be a `Stream`, but otherwise we should probably prefer `Future<List>`, or `Future<Iterable>`.
> 
> (`async*` `Stream`s also can have some surprising behavior. IIRC they don't interact well with `FakeAsync` for tests.)